td.vue/src/assets/threat-model-schema.js

```diff
@@ -0,0 +1,430 @@
+export const schema = {
```

Comment:
Hello @raghav1030 , here are some changes that will get the schema working for version 1.x threatmodels
Version 2.x threatmodels have a different schema so I will work on those later this week

```
% git diff td.vue/src/assets/threat-model-schema.js
diff --git a/td.vue/src/assets/threat-model-schema.js b/td.vue/src/assets/threat-model-schema.js
index 463fb247..abdd8fe9 100644
--- a/td.vue/src/assets/threat-model-schema.js
+++ b/td.vue/src/assets/threat-model-schema.js
@@ -1,13 +1,13 @@
 export const schema = {
-    "$schema": "https://json-schema.org/draft/2020-12/schema",
-    "$id": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://owasp.org/www-project-threat-dragon/assets/schemas/owasp.threat-dragon.schema.json",
     "title": "Threat Dragon model schema",
     "description": "The threat models used by OWASP Threat Dragon",
     "type": "object",
     "properties": {
       "version": {
         "description": "Threat Dragon version used in the model",
-        "type": "string"
+        "type": "string",
+        "maxLength": 5
       },
       "summary": {
         "description": "Threat model project meta-data",
@@ -94,7 +94,7 @@ export const schema = {
                 "version": {
                   "description": "Threat Dragon version used in the diagram",
                   "type": "string",
-                  "minLength": 5
+                  "maxLength": 5
                 },
                 "diagramJson": {
                   "description": "The data-flow diagram components",
@@ -425,6 +425,7 @@ export const schema = {
         },
         "required": [ "contributors", "diagrams" ]
       },
-      "required": [ "summary", "detail" ]
     },
-  }
\ No newline at end of file
+    "required": [ "summary", "detail" ],
+    additionalProperties: false
+}
\ No newline at end of file
```

Reply:
> Hello @raghav1030 , here are some changes that will get the schema working for version 1.x threatmodels Version 2.x threatmodels have a different schema so I will work on those later this week
> 
> ```
> % git diff td.vue/src/assets/threat-model-schema.js
> --- a/td.vue/src/assets/threat-model-schema.js
> +++ b/td.vue/src/assets/threat-model-schema.js
> @@ -1,13 +1,13 @@
>  export const schema = {
> -    "$schema": "https://json-schema.org/draft/2020-12/schema",
> -    "$id": "https://json-schema.org/draft/2020-12/schema",
> +    "$id": "https://owasp.org/www-project-threat-dragon/assets/schemas/owasp.threat-dragon.schema.json",
>      "title": "Threat Dragon model schema",
>      "description": "The threat models used by OWASP Threat Dragon",
>      "type": "object",
>      "properties": {
>        "version": {
>          "description": "Threat Dragon version used in the model",
> -        "type": "string"
> +        "type": "string",
> +        "maxLength": 5
>        },
>        "summary": {
>          "description": "Threat model project meta-data",
> @@ -94,7 +94,7 @@ export const schema = {
>                  "version": {
>                    "description": "Threat Dragon version used in the diagram",
>                    "type": "string",
> -                  "minLength": 5
> +                  "maxLength": 5
>                  },
>                  "diagramJson": {
>                    "description": "The data-flow diagram components",
> @@ -425,6 +425,7 @@ export const schema = {
>          },
>          "required": [ "contributors", "diagrams" ]
>        },
> -      "required": [ "summary", "detail" ]
>      },
> -  }
> \ No newline at end of file
> +    "required": [ "summary", "detail" ],
> +    additionalProperties: false
> +}
> \ No newline at end of file
> ```

Sure, I think this will fix the incorrect acceptance of malformed models.